src/test/java/org/generationcp/middleware/api/nametype/GermplasmNameTypeServiceImplIntegrationTest.java

```diff
@@ -0,0 +1,49 @@
+package org.generationcp.middleware.api.nametype;
+
+import org.apache.commons.lang3.RandomStringUtils;
+import org.generationcp.middleware.IntegrationTestBase;
+import org.generationcp.middleware.manager.DaoFactory;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.PageRequest;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+
+public class GermplasmNameTypeServiceImplIntegrationTest extends IntegrationTestBase {
+
+	private DaoFactory daoFactory;
+
+	@Autowired
+	GermplasmNameTypeService germplasmNameTypeService;
+
+	@Before
+	public void setUp() throws Exception {
+		this.daoFactory = new DaoFactory(this.sessionProvder);
+	}
+
+	@Test
+	public void testCreateNameType(){
+		final GermplasmNameTypeRequestDTO germplasmNameTypeRequestDTO = new GermplasmNameTypeRequestDTO();
+		germplasmNameTypeRequestDTO.setName("TEST" + RandomStringUtils.randomAlphabetic(10));
+		germplasmNameTypeRequestDTO.setCode("TEST" + RandomStringUtils.randomAlphabetic(10));
+		germplasmNameTypeRequestDTO.setDescription("Test Description" + RandomStringUtils.randomAlphabetic(10));
+		final Integer nameTypeId = this.germplasmNameTypeService.createNameType(germplasmNameTypeRequestDTO);
+		final Set<String> codes = new HashSet<>(Arrays.asList(germplasmNameTypeRequestDTO.getCode()));
+		final List<GermplasmNameTypeDTO> nameTypeRequestDTOs = this.germplasmNameTypeService.filterGermplasmNameTypes(codes);
+		Assert.assertThat(1, equalTo(nameTypeRequestDTOs.size()));
+		Assert.assertThat(nameTypeId, equalTo(nameTypeRequestDTOs.get(0).getId()));
+	}
+
+	@Test
+	public void testGetNameTypes() {
```

Comment:
In this scenario, what you are testing is the pagination I think? In that case, rename the function